### PR TITLE
Decouple Protocol-Def from Common-Def

### PR DESCRIPTION
**Pull request by anthony-murphy**
The only dependency between protocol-def and common-def was for eventing. I've re-implemented this without the need for importing the common types. This is type compatible, illustrated by the type tests.

### CODE REVIEW

File: common/lib/protocol-definitions/package.json [hunk 1]
```diff
@@ -36,9 +36,7 @@
 		"typetests:gen": "fluid-type-test-generator",
 		"typetests:prepare": "flub typetests --dir . --reset --previous --normalize"
 	},
-	"dependencies": {
-		"@fluidframework/common-definitions": "^1.0.0"
-	},
+	"dependencies": {},
```
Comment (Josmithr) [inline common/lib/protocol-definitions/package.json:39]: 🥳

**Review by Josmithr — APPROVED** [review]
Love this! Thanks for cleaning this up!

**Review by ChumpChief — COMMENTED** [review]
No real objection from me, but not sure I understand the motivation?  Are you saying it should be forbidden for protocol-definitions to depend on common-definitons (and if so, why is that)?

If so, probably remove the dependency allowance from layerInfo.json too.

**Comment by anthony-murphy** [comment]
> No real objection from me, but not sure I understand the motivation? Are you saying it should be forbidden for protocol-definitions to depend on common-definitons (and if so, why is that)?
> 
> If so, probably remove the dependency allowance from layerInfo.json too.

@ChumpChief  the comment was around odsp client, it was consuming a deprecated type from common-def. odsp client should not consume any deprecated types. i'm surpised the lint rule to prevent referencing deprecated types are not enabled.

**Comment by ChumpChief** [comment]
> > No real objection from me, but not sure I understand the motivation? Are you saying it should be forbidden for protocol-definitions to depend on common-definitons (and if so, why is that)?
> > If so, probably remove the dependency allowance from layerInfo.json too.
> 
> @ChumpChief the comment was around odsp client, it was consuming a deprecated type from common-def. odsp client should not consume any deprecated types. i'm surpised the lint rule to prevent referencing deprecated types are not enabled.

No I get that one, my comment is that I don't understand why it matters if protocol-def depends on common-def for eventing.

**Pull request merged**